public/app/features/explore/state/actions.ts

```diff
@@ -319,34 +361,13 @@ export function loadDatasource(exploreId: ExploreId, instance: DataSourceApi): T
     }
 
     // Check if queries can be imported from previously selected datasource
-    const queries = getState().explore[exploreId].modifiedQueries;
```

Comment:
Importing came after `instance.init();`. Have you checked if that makes a difference?

Reply:
Haven't noticed. Was there any particular reason to have queries imported after `init`?

Reply:
It's currently not relevant. Only Prometheus uses `init()`, but perhaps we should make that contract a bit clearer.

Reply:
what does init do? can it be moved to DatasourceSrv..get ?

Reply:
In case of Prometheus it loads rules. DatasourceSrv.get could perform the whole DS setup, init included after the DS is loaded. That would make sense to have DatasourceSrv return fully usable DS instance.